apis/apiextensions/fn/proto/v1beta1/run_function.proto

```diff
@@ -54,6 +54,12 @@ message RunFunctionRequest {
   // Optional input specific to this Function invocation. A JSON representation
   // of the 'input' block of the relevant entry in a Composition's pipeline.
   optional google.protobuf.Struct input = 4;
+
+  // Optional context. Crossplane may pass arbitary contextual information to a
+  // Function. A Function may also return context in its RunFunctionResponse,
+  // and that context will be passed to subsequent Functions. Crossplane
+  // discards all context returned by the last Function in the pipeline.
+  map<string, google.protobuf.Value> context = 5;
```

Comment:
```suggestion
  optional google.protobuf.Struct context = 5;
```

It occurs to me in retrospect that these are pretty much equivalent, given that a `google.protobuf.Struct` _is_ a map of string key fields to `google.protobuf.Value`. So this is semantically equivalent to what @phisco did in https://github.com/crossplane/crossplane/pull/4632.

Reply:
Which is what we were discussing here https://github.com/crossplane/crossplane/pull/4632#discussion_r1327686709 😅 however I ended up not making it just a map, but adding an explicit data field to the environment for the reasons I was mentioning above.

Reply:
Haha I somehow completely forgot about that conversation. My bad.

I think we're converging on something pretty close to what you already have in #4632 so I'll close this PR. Should have re-read it before I started working on this one. 🙃